client/asset/eth/eth.go

```diff
@@ -1453,34 +1455,37 @@ type gasEstimate struct {
 func (w *assetWallet) initGasEstimate(n int, initVer, redeemVer, redeemAssetID uint32) (est *gasEstimate, err error) {
 	est = new(gasEstimate)
 
-	g := w.gases(initVer)
-	if g == nil {
+	// Get the refund gas.
+	if g := w.gases(initVer); g == nil {
 		return nil, fmt.Errorf("no gas table")
+	} else { // scoping g
+		est.Refund = g.Refund
 	}
 
-	est.Refund = g.Refund
-
 	est.Swap, est.nSwap, _, err = w.swapGas(n, initVer)
 	if err != nil {
 		return nil, fmt.Errorf("error calculating swap gas: %w", err)
 	}
 
-	est.oneGas = est.Swap + est.Redeem
-	est.nGas = est.nSwap + est.nRedeem
+	est.oneGas = est.Swap
+	est.nGas = est.nSwap
 
 	if redeemW := w.wallet(redeemAssetID); redeemW != nil {
 		est.Redeem, est.nRedeem, err = redeemW.redeemGas(n, redeemVer)
 		if err != nil {
 			return nil, fmt.Errorf("error calculating fee-family redeem gas: %w", err)
 		}
+		est.oneGas += est.Redeem
+		est.nGas += est.nRedeem
```

Comment:
So this is supposed to be a bug fix, rightly pointed out by @buck54321 as doing nothing when it was above before getting est.Redeem/Redeemn, but isn't this inconsistent with the `oneSwap` that `swapGas` gets from `w.gases`?  Why should the init gas estimate include redeem gas at all?

Reply:
Oh lordy, I keep mixing up `estimateInitGas` and `initGasEstimate`.  NVM this concern about including Redeem gas.   This method is in fact just for funding, not tx authoring.